Populate new container ecs fields by docker module

MichaelKatsoulis opened the pull request:
<!-- Type of change
Please label this PR with one of the following labels, depending on the scope of your change:
- Bug
- Enhancement
- Breaking change
- Deprecation
- Cleanup
- Docs
-->

## What does this PR do?

<!-- Mandatory
Explain here the changes you made on the PR. Please explain the WHAT: patterns used, algorithms implemented, design architecture, message processing, etc.
-->

## Why is it important?

<!-- Mandatory
Explain here the WHY, or the rationale/motivation for the changes.
-->

After new container ECS fields get GA (https://github.com/elastic/ecs/pull/1747) Docker module need to start populating those fields.

The new fields are 
1. `container.cpu.usage`
2. `container.memory.usage `
3. `container.network.egress.bytes`
4. `container.network.ingress.bytes`
5. `container.disk.read.bytes`
6. `container.disk.write.bytes`

These fields map to docker fields:
1. `docker.cpu.total.norm.pct` by cpu metricset
2. `docker.memory.usage.pct` by memory metricset
3. `docker.network.outbound.bytes` by network metricset
4. `docker.network.inbound.bytes` by network metricset
5. `docker.diskio.read.bytes` by diskio metricset
6. `docker.diskio.write.bytes` by diskio metricset

respectively.

## Checklist

<!-- Mandatory
Add a checklist of things that are required to be reviewed in order to have the PR approved

List here all the items you have verified BEFORE sending this PR. Please DO NOT remove any item, striking through those that do not apply. (Just in case, strikethrough uses two tildes. ~~Scratch this.~~)
-->

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

<!-- Recommended
Add a checklist of things that are required to be reviewed in order to have the PR approved
-->
- [ ]

## How to test this PR locally

<!-- Recommended
Explain here how this PR will be tested by the reviewer: commands, dependencies, steps, etc.
-->


## Use cases

<!-- Recommended
Explain here the different behaviors that this PR introduces or modifies in this project, user roles, environment configuration, etc.

If you are familiar with Gherkin test scenarios, we recommend its usage: https://cucumber.io/docs/gherkin/reference/
-->

## Screenshots

<!-- Optional
Add here screenshots about how the project will be changed after the PR is applied. They could be related to web pages, terminal, etc, or any other image you consider important to be shared with the team.
-->
<img width="902" alt="mappings 1" src="https://user-images.githubusercontent.com/26270880/160580644-74c54dfc-585a-4d34-af94-23423bf67221.png">
<img width="915" alt="mappings 2" src="https://user-images.githubusercontent.com/26270880/160580651-77d6f901-eed2-4da9-8d51-63099e3fef37.png">
<img width="1776" alt="docker 5" src="https://user-images.githubusercontent.com/26270880/160580669-6a76ee3b-6428-401b-b034-fa81a6c90494.png">
<img width="1777" alt="docker 4" src="https://user-images.githubusercontent.com/26270880/160580706-7d294570-f79d-49c7-be33-0f0dc034ae9f.png">
<img width="1781" alt="docker 3" src="https://user-images.githubusercontent.com/26270880/160580750-221cce4b-c54d-4a93-a5e1-750674754a57.png">
<img width="1782" alt="docker 1" src="https://user-images.githubusercontent.com/26270880/160580787-f9a6169c-9458-461f-b94d-9935218df72d.png">
<img width="1770" alt="docker 2" src="https://user-images.githubusercontent.com/26270880/160580824-3209df21-c95f-460e-89c3-c6b6df6bd246.png">

elasticmachine commented:
Pinging @elastic/integrations (Team:Integrations)

elasticmachine commented:
## :broken_heart: Tests Failed


<!-- BUILD BADGES-->
> _the below badges are clickable and redirect to their specific view in the CI or DOCS_
[![Pipeline View](https://img.shields.io/badge/pipeline-pipeline%20-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2FPR-30399/detail/PR-30399/1//pipeline) [![Test View](https://img.shields.io/badge/test-test-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2FPR-30399/detail/PR-30399/1//tests) [![Changes](https://img.shields.io/badge/changes-changes-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2FPR-30399/detail/PR-30399/1//changes) [![Artifacts](https://img.shields.io/badge/artifacts-artifacts-yellow)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2FPR-30399/detail/PR-30399/1//artifacts) [![preview](https://img.shields.io/badge/docs-preview-yellowgreen)](http://beats_30399.docs-preview.app.elstc.co/diff)  [![preview](https://img.shields.io/badge/elastic-observability-blue)](https://ci-stats.elastic.co/app/apm/services/beats-ci/transactions/view?rangeFrom=2022-02-15T09:36:10.088Z&rangeTo=2022-02-15T09:56:10.088Z&transactionName=Beats/beats/PR-{number}&transactionType=job&latencyAggregationType=avg&traceId=bc2132756050350fc523ff81f28492b1&transactionId=c1fe7d240353badc)

<!-- BUILD SUMMARY-->
<details><summary>Expand to view the summary</summary>
<p>

#### Build stats




* Start Time: 2022-02-15T09:46:10.088+0000


* Duration: 42 min 49 sec




#### Test stats :test_tube:

| Test         | Results                         |
| ------------ | :-----------------------------: |
| Failed       | 7  |
| Passed       | 2411  |
| Skipped      | 165 |
| Total        | 2583   |


</p>
</details>

<!-- TEST RESULTS IF ANY-->

  ### Test errors [![7](https://img.shields.io/badge/7%20-red)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2FPR-30399/detail/PR-30399/1//tests)
<details><summary>Expand to view the tests failures</summary><p>
    
    
##### `Build&Test / metricbeat-goIntegTest / TestMemoryService_GetMemoryStats – github.com/elastic/beats/v7/metricbeat/module/docker/memory`


<ul>
<details><summary>Expand to view the error details</summary><p>

```
 Failed 
 ```
</p></details>
<details><summary>Expand to view the stacktrace</summary><p>

```
 === RUN   TestMemoryService_GetMemoryStats
    memory_test.go:108: 
        	Error Trace:	memory_test.go:108
        	Error:      	Not equal: 
        	            	expected: {"container":{"id":"containerID","image":{"name":"image"},"name":"name1","runtime":"docker"},"docker":{"container":{"labels":{"label1":"val1","label2":{"foo":"val3","value":"val2"}}}}}
        	            	actual  : {"container":{"id":"containerID","image":{"name":"image"},"memory":{"usage":0.5},"name":"name1","runtime":"docker"},"docker":{"container":{"labels":{"label1":"val1","label2":{"foo":"val3","value":"val2"}}}}}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,3 +1,3 @@
        	            	 (common.MapStr) (len=2) {
        	            	- (string) (len=9) "container": (common.MapStr) (len=4) {
        	            	+ (string) (len=9) "container": (common.MapStr) (len=5) {
        	            	   (string) (len=2) "id": (string) (len=11) "containerID",
        	            	@@ -5,2 +5,5 @@
        	            	    (string) (len=4) "name": (string) (len=5) "image"
        	            	+  },
        	            	+  (string) (len=6) "memory": (common.MapStr) (len=1) {
        	            	+   (string) (len=5) "usage": (float64) 0.5
        	            	   },
        	Test:       	TestMemoryService_GetMemoryStats
--- FAIL: TestMemoryService_GetMemoryStats (0.00s)
 
 ```
</p></details>
</ul>
    
##### `Build&Test / metricbeat-unitTest / TestMemoryService_GetMemoryStats – github.com/elastic/beats/v7/metricbeat/module/docker/memory`


<ul>
<details><summary>Expand to view the error details</summary><p>

```
 Failed 
 ```
</p></details>
<details><summary>Expand to view the stacktrace</summary><p>

```
 === RUN   TestMemoryService_GetMemoryStats
    memory_test.go:108: 
        	Error Trace:	memory_test.go:108
        	Error:      	Not equal: 
        	            	expected: {"container":{"id":"containerID","image":{"name":"image"},"name":"name1","runtime":"docker"},"docker":{"container":{"labels":{"label1":"val1","label2":{"foo":"val3","value":"val2"}}}}}
        	            	actual  : {"container":{"id":"containerID","image":{"name":"image"},"memory":{"usage":0.5},"name":"name1","runtime":"docker"},"docker":{"container":{"labels":{"label1":"val1","label2":{"foo":"val3","value":"val2"}}}}}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,3 +1,3 @@
        	            	 (common.MapStr) (len=2) {
        	            	- (string) (len=9) "container": (common.MapStr) (len=4) {
        	            	+ (string) (len=9) "container": (common.MapStr) (len=5) {
        	            	   (string) (len=2) "id": (string) (len=11) "containerID",
        	            	@@ -5,2 +5,5 @@
        	            	    (string) (len=4) "name": (string) (len=5) "image"
        	            	+  },
        	            	+  (string) (len=6) "memory": (common.MapStr) (len=1) {
        	            	+   (string) (len=5) "usage": (float64) 0.5
        	            	   },
        	Test:       	TestMemoryService_GetMemoryStats
--- FAIL: TestMemoryService_GetMemoryStats (0.00s)
 
 ```
</p></details>
</ul>
    
##### `Build&Test / metricbeat-windows-windows-2019 / TestMemoryService_GetMemoryStats – github.com/elastic/beats/v7/metricbeat/module/docker/memory`


<ul>
<details><summary>Expand to view the error details</summary><p>

```
 Failed 
 ```
</p></details>
<details><summary>Expand to view the stacktrace</summary><p>

```
 === RUN   TestMemoryService_GetMemoryStats
    memory_test.go:108: 
        	Error Trace:	memory_test.go:108
        	Error:      	Not equal: 
        	            	expected: {"container":{"id":"containerID","image":{"name":"image"},"name":"name1","runtime":"docker"},"docker":{"container":{"labels":{"label1":"val1","label2":{"foo":"val3","value":"val2"}}}}}
        	            	actual  : {"container":{"id":"containerID","image":{"name":"image"},"memory":{"usage":0.5},"name":"name1","runtime":"docker"},"docker":{"container":{"labels":{"label1":"val1","label2":{"foo":"val3","value":"val2"}}}}}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,3 +1,3 @@
        	            	 (common.MapStr) (len=2) {
        	            	- (string) (len=9) "container": (common.MapStr) (len=4) {
        	            	+ (string) (len=9) "container": (common.MapStr) (len=5) {
        	            	   (string) (len=2) "id": (string) (len=11) "containerID",
        	            	@@ -5,2 +5,5 @@
        	            	    (string) (len=4) "name": (string) (len=5) "image"
        	            	+  },
        	            	+  (string) (len=6) "memory": (common.MapStr) (len=1) {
        	            	+   (string) (len=5) "usage": (float64) 0.5
        	            	   },
        	Test:       	TestMemoryService_GetMemoryStats
--- FAIL: TestMemoryService_GetMemoryStats (0.00s)
 
 ```
</p></details>
</ul>
    
##### `Build&Test / metricbeat-pythonIntegTest / test_cpu_fields – metricbeat.module.docker.test_docker.Test`


<ul>
<details><summary>Expand to view the error details</summary><p>

```
 Exception: Key 'container.cpu.usage' found in event ({'@timestamp': '2022-02-15T10:19:37.778Z', 'host': {'name': 'beats-ci-immutable-ubuntu-1804-1644919609072491002'}, 'agent': {'id': '23b4f0b9-c60a-4bfb-aff1-ecf05d818801', 'name': 'beats-ci-immutable-ubuntu-1804-1644919609072491002', 'type': 'metricbeat', 'version': '8.2.0', 'ephemeral_id': '88321f6e-f602-48bd-9c97-e46db07ddee3'}, 'ecs': {'version': '8.0.0'}, 'container': {'image': {'name': 'alpine'}, 'runtime': 'docker', 'cpu': {'usage': 0.0034675203007518795}, 'id': 'ceb909223b04fec3cd3222e933d2e45fb2b562b91260cd628b63d206a41bbbbd', 'name': 'docker_7918a246f6f8_docker_1'}, 'docker': {'container': {}, 'cpu': {'total': {'norm': {'pct': 0.0034675203007518795}, 'pct': 0.027740162406015036}, 'kernel': {'ticks': 40000000, 'pct': 0, 'norm': {'pct': 0}}, 'user': {'ticks': 120000000, 'pct': 0.020050125313283207, 'norm': {'pct': 0.002506265664160401}}, 'system': {'pct': 8, 'norm': {'pct': 1}, 'ticks': 5902350000000}}}, 'event': {'dataset': 'docker.cpu', 'module': 'docker', 'duration': 2013738333}, 'metricset': {'name': 'cpu', 'period': 10000}, 'service': {'address': 'unix:///var/run/docker.sock', 'type': 'docker'}}) is not documented! 
 ```
</p></details>
<details><summary>Expand to view the stacktrace</summary><p>

```
 self = <test_docker.Test testMethod=test_cpu_fields>

    @unittest.skipUnless(metricbeat.INTEGRATION_TESTS, "integration test")
    def test_cpu_fields(self):
        """
        test cpu fields
        """
        self.render_config_template(modules=[{
            "name": "docker",
            "metricsets": ["cpu"],
            "hosts": ["unix:///var/run/docker.sock"],
            "period": "10s"
        }])
    
        proc = self.start_beat()
        self.wait_until(lambda: self.output_lines() > 0, max_timeout=30)
        proc.check_kill_and_wait()
        self.assert_no_logged_warnings(["Container stopped when recovering stats",
                                        "An error occurred while getting docker stats"])
    
        output = self.read_output_json()
        evt = output[0]
    
        evt = self.remove_labels(evt)
    
        if 'core' in evt["docker"]["cpu"]:
            del evt["docker"]["cpu"]["core"]
    
>       self.assert_fields_are_documented(evt)

module/docker/test_docker.py:63: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <test_docker.Test testMethod=test_cpu_fields>
evt = {'@timestamp': '2022-02-15T10:19:37.778Z', 'agent': {'ephemeral_id': '88321f6e-f602-48bd-9c97-e46db07ddee3', 'id': '23...162406015036}, 'user': {'norm': {'pct': 0.002506265664160401}, 'pct': 0.020050125313283207, 'ticks': 120000000}}}, ...}

    def assert_fields_are_documented(self, evt):
        """
        Assert that all keys present in evt are documented in fields.yml.
        This reads from the global fields.yml, means `make collect` has to be run before the check.
        """
        expected_fields, dict_fields, aliases = self.load_fields()
        flat = self.flatten_object(evt, dict_fields)
    
        def field_pattern_match(pattern, key):
            pattern_fields = pattern.split(".")
            key_fields = key.split(".")
            if len(pattern_fields) != len(key_fields):
                return False
            for i in range(len(pattern_fields)):
                if pattern_fields[i] == "*":
                    continue
                if pattern_fields[i] != key_fields[i]:
                    return False
            return True
    
        def is_documented(key, docs):
            if key in docs:
                return True
            for pattern in (f for f in docs if "*" in f):
                if field_pattern_match(pattern, key):
                    return True
            return False
    
        for key in flat.keys():
            metaKey = key.startswith('@metadata.')
            # Range keys as used in 'date_range' etc will not have docs of course
            isRangeKey = key.split('.')[-1] in ['gte', 'gt', 'lte', 'lt']
            if not(is_documented(key, expected_fields) or metaKey or isRangeKey):
>               raise Exception("Key '{}' found in event ({}) is not documented!".format(key, str(evt)))
E               Exception: Key 'container.cpu.usage' found in event ({'@timestamp': '2022-02-15T10:19:37.778Z', 'host': {'name': 'beats-ci-immutable-ubuntu-1804-1644919609072491002'}, 'agent': {'id': '23b4f0b9-c60a-4bfb-aff1-ecf05d818801', 'name': 'beats-ci-immutable-ubuntu-1804-1644919609072491002', 'type': 'metricbeat', 'version': '8.2.0', 'ephemeral_id': '88321f6e-f602-48bd-9c97-e46db07ddee3'}, 'ecs': {'version': '8.0.0'}, 'container': {'image': {'name': 'alpine'}, 'runtime': 'docker', 'cpu': {'usage': 0.0034675203007518795}, 'id': 'ceb909223b04fec3cd3222e933d2e45fb2b562b91260cd628b63d206a41bbbbd', 'name': 'docker_7918a246f6f8_docker_1'}, 'docker': {'container': {}, 'cpu': {'total': {'norm': {'pct': 0.0034675203007518795}, 'pct': 0.027740162406015036}, 'kernel': {'ticks': 40000000, 'pct': 0, 'norm': {'pct': 0}}, 'user': {'ticks': 120000000, 'pct': 0.020050125313283207, 'norm': {'pct': 0.002506265664160401}}, 'system': {'pct': 8, 'norm': {'pct': 1}, 'ticks': 5902350000000}}}, 'event': {'dataset': 'docker.cpu', 'module': 'docker', 'duration': 2013738333}, 'metricset': {'name': 'cpu', 'period': 10000}, 'service': {'address': 'unix:///var/run/docker.sock', 'type': 'docker'}}) is not documented!

../libbeat/tests/system/beat/beat.py:779: Exception 
 ```
</p></details>
</ul>
    
##### `Build&Test / metricbeat-pythonIntegTest / test_diskio_fields – metricbeat.module.docker.test_docker.Test`


<ul>
<details><summary>Expand to view the error details</summary><p>

```
 Exception: Key 'container.disk.read.bytes' found in event ({'@timestamp': '2022-02-15T10:19:40.623Z', 'event': {'duration': 1179426491, 'dataset': 'docker.diskio', 'module': 'docker'}, 'container': {'runtime': 'docker', 'disk': {'read': {'bytes': 12288}, 'write': {'bytes': 1110016}}, 'id': '003c79f68aaddb2beb72d51ca0ffa1feb2038cf68692c75b22da4d8efc479d6d', 'name': 'metricbeat_8_2_0_f4a863c246-snapshot_beat_run_1', 'image': {'name': 'metricbeat_8_2_0_f4a863c246-snapshot_beat'}}, 'docker': {'container': {}, 'diskio': {'read': {'queued': 0, 'ops': 3, 'bytes': 12288, 'rate': 0, 'service_time': 0, 'wait_time': 0}, 'write': {'queued': 0, 'ops': 260, 'bytes': 1110016, 'rate': 0, 'service_time': 0, 'wait_time': 0}, 'summary': {'ops': 263, 'bytes': 1122304, 'rate': 0, 'service_time': 0, 'wait_time': 0, 'queued': 0}}}, 'host': {'name': 'beats-ci-immutable-ubuntu-1804-1644919609072491002'}, 'agent': {'type': 'metricbeat', 'version': '8.2.0', 'ephemeral_id': '30e36880-8492-4c28-b83f-fda6016f760c', 'id': 'baf171ed-e4e3-466e-b1e6-6e10868ace73', 'name': 'beats-ci-immutable-ubuntu-1804-1644919609072491002'}, 'ecs': {'version': '8.0.0'}, 'metricset': {'name': 'diskio', 'period': 10000}, 'service': {'address': 'unix:///var/run/docker.sock', 'type': 'docker'}}) is not documented! 
 ```
</p></details>
<details><summary>Expand to view the stacktrace</summary><p>

```
 self = <test_docker.Test testMethod=test_diskio_fields>

    @unittest.skipUnless(metricbeat.INTEGRATION_TESTS, "integration test")
    def test_diskio_fields(self):
        """
        test diskio fields
        """
        self.render_config_template(modules=[{
            "name": "docker",
            "metricsets": ["diskio"],
            "hosts": ["unix:///var/run/docker.sock"],
            "period": "10s"
        }])
    
        proc = self.start_beat()
        self.wait_until(lambda: self.output_lines() > 0, max_timeout=30)
        proc.check_kill_and_wait()
        self.assert_no_logged_warnings(["Container stopped when recovering stats",
                                        "An error occurred while getting docker stats"])
    
        output = self.read_output_json()
        evt = output[0]
    
        evt = self.remove_labels(evt)
    
>       self.assert_fields_are_documented(evt)

module/docker/test_docker.py:88: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <test_docker.Test testMethod=test_diskio_fields>
evt = {'@timestamp': '2022-02-15T10:19:40.623Z', 'agent': {'ephemeral_id': '30e36880-8492-4c28-b83f-fda6016f760c', 'id': 'ba... 'ops': 263, 'queued': 0, 'rate': 0, ...}, 'write': {'bytes': 1110016, 'ops': 260, 'queued': 0, 'rate': 0, ...}}}, ...}

    def assert_fields_are_documented(self, evt):
        """
        Assert that all keys present in evt are documented in fields.yml.
        This reads from the global fields.yml, means `make collect` has to be run before the check.
        """
        expected_fields, dict_fields, aliases = self.load_fields()
        flat = self.flatten_object(evt, dict_fields)
    
        def field_pattern_match(pattern, key):
            pattern_fields = pattern.split(".")
            key_fields = key.split(".")
            if len(pattern_fields) != len(key_fields):
                return False
            for i in range(len(pattern_fields)):
                if pattern_fields[i] == "*":
                    continue
                if pattern_fields[i] != key_fields[i]:
                    return False
            return True
    
        def is_documented(key, docs):
            if key in docs:
                return True
            for pattern in (f for f in docs if "*" in f):
                if field_pattern_match(pattern, key):
                    return True
            return False
    
        for key in flat.keys():
            metaKey = key.startswith('@metadata.')
            # Range keys as used in 'date_range' etc will not have docs of course
            isRangeKey = key.split('.')[-1] in ['gte', 'gt', 'lte', 'lt']
            if not(is_documented(key, expected_fields) or metaKey or isRangeKey):
>               raise Exception("Key '{}' found in event ({}) is not documented!".format(key, str(evt)))
E               Exception: Key 'container.disk.read.bytes' found in event ({'@timestamp': '2022-02-15T10:19:40.623Z', 'event': {'duration': 1179426491, 'dataset': 'docker.diskio', 'module': 'docker'}, 'container': {'runtime': 'docker', 'disk': {'read': {'bytes': 12288}, 'write': {'bytes': 1110016}}, 'id': '003c79f68aaddb2beb72d51ca0ffa1feb2038cf68692c75b22da4d8efc479d6d', 'name': 'metricbeat_8_2_0_f4a863c246-snapshot_beat_run_1', 'image': {'name': 'metricbeat_8_2_0_f4a863c246-snapshot_beat'}}, 'docker': {'container': {}, 'diskio': {'read': {'queued': 0, 'ops': 3, 'bytes': 12288, 'rate': 0, 'service_time': 0, 'wait_time': 0}, 'write': {'queued': 0, 'ops': 260, 'bytes': 1110016, 'rate': 0, 'service_time': 0, 'wait_time': 0}, 'summary': {'ops': 263, 'bytes': 1122304, 'rate': 0, 'service_time': 0, 'wait_time': 0, 'queued': 0}}}, 'host': {'name': 'beats-ci-immutable-ubuntu-1804-1644919609072491002'}, 'agent': {'type': 'metricbeat', 'version': '8.2.0', 'ephemeral_id': '30e36880-8492-4c28-b83f-fda6016f760c', 'id': 'baf171ed-e4e3-466e-b1e6-6e10868ace73', 'name': 'beats-ci-immutable-ubuntu-1804-1644919609072491002'}, 'ecs': {'version': '8.0.0'}, 'metricset': {'name': 'diskio', 'period': 10000}, 'service': {'address': 'unix:///var/run/docker.sock', 'type': 'docker'}}) is not documented!

../libbeat/tests/system/beat/beat.py:779: Exception 
 ```
</p></details>
</ul>
    
##### `Build&Test / metricbeat-pythonIntegTest / test_memory_fields – metricbeat.module.docker.test_docker.Test`


<ul>
<details><summary>Expand to view the error details</summary><p>

```
 Exception: Key 'container.memory.usage' found in event ({'@timestamp': '2022-02-15T10:19:45.072Z', 'host': {'name': 'beats-ci-immutable-ubuntu-1804-1644919609072491002'}, 'docker': {'container': {}, 'memory': {'stats': {'total_inactive_file': 0, 'total_pgmajfault': 0, 'total_rss_huge': 0, 'active_anon': 0, 'mapped_file': 0, 'pgmajfault': 0, 'pgpgin': 6963, 'total_mapped_file': 0, 'total_pgpgout': 6940, 'active_file': 0, 'hierarchical_memsw_limit': 0, 'total_cache': 0, 'inactive_file': 0, 'pgpgout': 6940, 'total_rss': 0, 'unevictable': 0, 'total_unevictable': 0, 'total_writeback': 0, 'rss_huge': 0, 'total_pgfault': 9438, 'total_active_anon': 0, 'total_pgpgin': 6963, 'hierarchical_memory_limit': 9.223372036854772e+18, 'rss': 0, 'inactive_anon': 0, 'total_active_file': 0, 'cache': 0, 'dirty': 0, 'total_inactive_anon': 0, 'writeback': 0, 'pgfault': 9438, 'total_dirty': 0}, 'fail': {'count': 0}, 'limit': 54757818368, 'rss': {'pct': 0, 'total': 0}, 'usage': {'total': 2281472, 'pct': 4.166477168004328e-05, 'max': 5029888}}}, 'event': {'duration': 2015232107, 'dataset': 'docker.memory', 'module': 'docker'}, 'metricset': {'name': 'memory', 'period': 10000}, 'service': {'address': 'unix:///var/run/docker.sock', 'type': 'docker'}, 'container': {'name': 'docker_7918a246f6f8_docker_1', 'image': {'name': 'alpine'}, 'runtime': 'docker', 'memory': {'usage': 4.166477168004328e-05}, 'id': 'ceb909223b04fec3cd3222e933d2e45fb2b562b91260cd628b63d206a41bbbbd'}, 'agent': {'type': 'metricbeat', 'version': '8.2.0', 'ephemeral_id': '76556964-c705-4974-9d64-bdabf2937427', 'id': 'cba639e3-b884-4ac3-a93c-70f164643f22', 'name': 'beats-ci-immutable-ubuntu-1804-1644919609072491002'}, 'ecs': {'version': '8.0.0'}}) is not documented! 
 ```
</p></details>
<details><summary>Expand to view the stacktrace</summary><p>

```
 self = <test_docker.Test testMethod=test_memory_fields>

    @unittest.skipUnless(metricbeat.INTEGRATION_TESTS, "integration test")
    def test_memory_fields(self):
        """
        test memory fields
        """
        self.render_config_template(modules=[{
            "name": "docker",
            "metricsets": ["memory"],
            "hosts": ["unix:///var/run/docker.sock"],
            "period": "10s"
        }])
    
        proc = self.start_beat()
        self.wait_until(lambda: self.output_lines() > 0, max_timeout=30)
        proc.check_kill_and_wait()
        self.assert_no_logged_warnings(["Container stopped when recovering stats",
                                        "An error occurred while getting docker stats"])
    
        output = self.read_output_json()
        evt = output[0]
    
        evt = self.remove_labels(evt)
>       self.assert_fields_are_documented(evt)

module/docker/test_docker.py:135: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <test_docker.Test testMethod=test_memory_fields>
evt = {'@timestamp': '2022-02-15T10:19:45.072Z', 'agent': {'ephemeral_id': '76556964-c705-4974-9d64-bdabf2937427', 'id': 'cb... 'rss': {'pct': 0, 'total': 0}, 'stats': {'active_anon': 0, 'active_file': 0, 'cache': 0, 'dirty': 0, ...}, ...}}, ...}

    def assert_fields_are_documented(self, evt):
        """
        Assert that all keys present in evt are documented in fields.yml.
        This reads from the global fields.yml, means `make collect` has to be run before the check.
        """
        expected_fields, dict_fields, aliases = self.load_fields()
        flat = self.flatten_object(evt, dict_fields)
    
        def field_pattern_match(pattern, key):
            pattern_fields = pattern.split(".")
            key_fields = key.split(".")
            if len(pattern_fields) != len(key_fields):
                return False
            for i in range(len(pattern_fields)):
                if pattern_fields[i] == "*":
                    continue
                if pattern_fields[i] != key_fields[i]:
                    return False
            return True
    
        def is_documented(key, docs):
            if key in docs:
                return True
            for pattern in (f for f in docs if "*" in f):
                if field_pattern_match(pattern, key):
                    return True
            return False
    
        for key in flat.keys():
            metaKey = key.startswith('@metadata.')
            # Range keys as used in 'date_range' etc will not have docs of course
            isRangeKey = key.split('.')[-1] in ['gte', 'gt', 'lte', 'lt']
            if not(is_documented(key, expected_fields) or metaKey or isRangeKey):
>               raise Exception("Key '{}' found in event ({}) is not documented!".format(key, str(evt)))
E               Exception: Key 'container.memory.usage' found in event ({'@timestamp': '2022-02-15T10:19:45.072Z', 'host': {'name': 'beats-ci-immutable-ubuntu-1804-1644919609072491002'}, 'docker': {'container': {}, 'memory': {'stats': {'total_inactive_file': 0, 'total_pgmajfault': 0, 'total_rss_huge': 0, 'active_anon': 0, 'mapped_file': 0, 'pgmajfault': 0, 'pgpgin': 6963, 'total_mapped_file': 0, 'total_pgpgout': 6940, 'active_file': 0, 'hierarchical_memsw_limit': 0, 'total_cache': 0, 'inactive_file': 0, 'pgpgout': 6940, 'total_rss': 0, 'unevictable': 0, 'total_unevictable': 0, 'total_writeback': 0, 'rss_huge': 0, 'total_pgfault': 9438, 'total_active_anon': 0, 'total_pgpgin': 6963, 'hierarchical_memory_limit': 9.223372036854772e+18, 'rss': 0, 'inactive_anon': 0, 'total_active_file': 0, 'cache': 0, 'dirty': 0, 'total_inactive_anon': 0, 'writeback': 0, 'pgfault': 9438, 'total_dirty': 0}, 'fail': {'count': 0}, 'limit': 54757818368, 'rss': {'pct': 0, 'total': 0}, 'usage': {'total': 2281472, 'pct': 4.166477168004328e-05, 'max': 5029888}}}, 'event': {'duration': 2015232107, 'dataset': 'docker.memory', 'module': 'docker'}, 'metricset': {'name': 'memory', 'period': 10000}, 'service': {'address': 'unix:///var/run/docker.sock', 'type': 'docker'}, 'container': {'name': 'docker_7918a246f6f8_docker_1', 'image': {'name': 'alpine'}, 'runtime': 'docker', 'memory': {'usage': 4.166477168004328e-05}, 'id': 'ceb909223b04fec3cd3222e933d2e45fb2b562b91260cd628b63d206a41bbbbd'}, 'agent': {'type': 'metricbeat', 'version': '8.2.0', 'ephemeral_id': '76556964-c705-4974-9d64-bdabf2937427', 'id': 'cba639e3-b884-4ac3-a93c-70f164643f22', 'name': 'beats-ci-immutable-ubuntu-1804-1644919609072491002'}, 'ecs': {'version': '8.0.0'}}) is not documented!

../libbeat/tests/system/beat/beat.py:779: Exception 
 ```
</p></details>
</ul>
    
##### `Build&Test / metricbeat-pythonIntegTest / test_network_fields – metricbeat.module.docker.test_docker.Test`


<ul>
<details><summary>Expand to view the error details</summary><p>

```
 Exception: Key 'container.network.ingress.bytes' found in event ({'@timestamp': '2022-02-15T10:19:47.850Z', 'agent': {'id': '3b072547-6e54-42e6-a76a-6adf854cec6c', 'name': 'beats-ci-immutable-ubuntu-1804-1644919609072491002', 'type': 'metricbeat', 'version': '8.2.0', 'ephemeral_id': 'b476cbc0-4751-4715-804d-1c75ca5af2d7'}, 'container': {'name': 'docker_7918a246f6f8_docker_1', 'image': {'name': 'alpine'}, 'runtime': 'docker', 'network': {'ingress': {'bytes': 1182}, 'egress': {'bytes': 0}}, 'id': 'ceb909223b04fec3cd3222e933d2e45fb2b562b91260cd628b63d206a41bbbbd'}, 'docker': {'container': {}, 'network': {'outbound': {'errors': 0, 'packets': 0, 'bytes': 0, 'dropped': 0}, 'interface': 'eth0', 'inbound': {'packets': 13, 'bytes': 1182, 'dropped': 0, 'errors': 0}}}, 'event': {'dataset': 'docker.network', 'module': 'docker', 'duration': 1249322306}, 'metricset': {'name': 'network', 'period': 10000}, 'service': {'type': 'docker', 'address': 'unix:///var/run/docker.sock'}, 'ecs': {'version': '8.0.0'}, 'host': {'name': 'beats-ci-immutable-ubuntu-1804-1644919609072491002'}}) is not documented! 
 ```
</p></details>
<details><summary>Expand to view the stacktrace</summary><p>

```
 self = <test_docker.Test testMethod=test_network_fields>

    @unittest.skipUnless(metricbeat.INTEGRATION_TESTS, "integration test")
    def test_network_fields(self):
        """
        test network fields
        """
        self.render_config_template(modules=[{
            "name": "docker",
            "metricsets": ["network"],
            "hosts": ["unix:///var/run/docker.sock"],
            "period": "10s"
        }])
    
        proc = self.start_beat()
        self.wait_until(lambda: self.output_lines() > 0, max_timeout=30)
        proc.check_kill_and_wait()
        self.assert_no_logged_warnings(["Container stopped when recovering stats",
                                        "An error occurred while getting docker stats"])
    
        output = self.read_output_json()
        evt = output[0]
    
        evt = self.remove_labels(evt)
>       self.assert_fields_are_documented(evt)

module/docker/test_docker.py:159: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <test_docker.Test testMethod=test_network_fields>
evt = {'@timestamp': '2022-02-15T10:19:47.850Z', 'agent': {'ephemeral_id': 'b476cbc0-4751-4715-804d-1c75ca5af2d7', 'id': '3b...ors': 0, 'packets': 13}, 'interface': 'eth0', 'outbound': {'bytes': 0, 'dropped': 0, 'errors': 0, 'packets': 0}}}, ...}

    def assert_fields_are_documented(self, evt):
        """
        Assert that all keys present in evt are documented in fields.yml.
        This reads from the global fields.yml, means `make collect` has to be run before the check.
        """
        expected_fields, dict_fields, aliases = self.load_fields()
        flat = self.flatten_object(evt, dict_fields)
    
        def field_pattern_match(pattern, key):
            pattern_fields = pattern.split(".")
            key_fields = key.split(".")
            if len(pattern_fields) != len(key_fields):
                return False
            for i in range(len(pattern_fields)):
                if pattern_fields[i] == "*":
                    continue
                if pattern_fields[i] != key_fields[i]:
                    return False
            return True
    
        def is_documented(key, docs):
            if key in docs:
                return True
            for pattern in (f for f in docs if "*" in f):
                if field_pattern_match(pattern, key):
                    return True
            return False
    
        for key in flat.keys():
            metaKey = key.startswith('@metadata.')
            # Range keys as used in 'date_range' etc will not have docs of course
            isRangeKey = key.split('.')[-1] in ['gte', 'gt', 'lte', 'lt']
            if not(is_documented(key, expected_fields) or metaKey or isRangeKey):
>               raise Exception("Key '{}' found in event ({}) is not documented!".format(key, str(evt)))
E               Exception: Key 'container.network.ingress.bytes' found in event ({'@timestamp': '2022-02-15T10:19:47.850Z', 'agent': {'id': '3b072547-6e54-42e6-a76a-6adf854cec6c', 'name': 'beats-ci-immutable-ubuntu-1804-1644919609072491002', 'type': 'metricbeat', 'version': '8.2.0', 'ephemeral_id': 'b476cbc0-4751-4715-804d-1c75ca5af2d7'}, 'container': {'name': 'docker_7918a246f6f8_docker_1', 'image': {'name': 'alpine'}, 'runtime': 'docker', 'network': {'ingress': {'bytes': 1182}, 'egress': {'bytes': 0}}, 'id': 'ceb909223b04fec3cd3222e933d2e45fb2b562b91260cd628b63d206a41bbbbd'}, 'docker': {'container': {}, 'network': {'outbound': {'errors': 0, 'packets': 0, 'bytes': 0, 'dropped': 0}, 'interface': 'eth0', 'inbound': {'packets': 13, 'bytes': 1182, 'dropped': 0, 'errors': 0}}}, 'event': {'dataset': 'docker.network', 'module': 'docker', 'duration': 1249322306}, 'metricset': {'name': 'network', 'period': 10000}, 'service': {'type': 'docker', 'address': 'unix:///var/run/docker.sock'}, 'ecs': {'version': '8.0.0'}, 'host': {'name': 'beats-ci-immutable-ubuntu-1804-1644919609072491002'}}) is not documented!

../libbeat/tests/system/beat/beat.py:779: Exception 
 ```
</p></details>
</ul>
    
</p></details>


<!-- STEPS ERRORS IF ANY -->











### Steps errors [![13](https://img.shields.io/badge/13%20-red)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2FPR-30399/detail/PR-30399/1//pipeline)
  <details><summary>Expand to view the steps failures</summary>
  <p>


> Show only the first 10 steps failures

  
  
  
  
##### `metricbeat-goIntegTest - mage goIntegTest`
<ul>
<li>Took 3 min 44 sec . View more details <a href="https://beats-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/Beats/pipelines/beats/pipelines/PR-30399/runs/1/steps/3299/log/?start=0">here</a></li>
<li>Description: <code>mage goIntegTest</code></l1>
</ul>
  
  
  
  
##### `metricbeat-goIntegTest - mage goIntegTest`
<ul>
<li>Took 0 min 28 sec . View more details <a href="https://beats-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/Beats/pipelines/beats/pipelines/PR-30399/runs/1/steps/3483/log/?start=0">here</a></li>
<li>Description: <code>mage goIntegTest</code></l1>
</ul>
  
  
  
  
##### `metricbeat-goIntegTest - mage goIntegTest`
<ul>
<li>Took 0 min 27 sec . View more details <a href="https://beats-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/Beats/pipelines/beats/pipelines/PR-30399/runs/1/steps/3495/log/?start=0">here</a></li>
<li>Description: <code>mage goIntegTest</code></l1>
</ul>
  
  
  
  
##### `metricbeat-pythonIntegTest - mage pythonIntegTest`
<ul>
<li>Took 5 min 19 sec . View more details <a href="https://beats-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/Beats/pipelines/beats/pipelines/PR-30399/runs/1/steps/3331/log/?start=0">here</a></li>
<li>Description: <code>mage pythonIntegTest</code></l1>
</ul>
  
  
  
  
##### `metricbeat-pythonIntegTest - mage pythonIntegTest`
<ul>
<li>Took 2 min 41 sec . View more details <a href="https://beats-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/Beats/pipelines/beats/pipelines/PR-30399/runs/1/steps/3757/log/?start=0">here</a></li>
<li>Description: <code>mage pythonIntegTest</code></l1>
</ul>
  
  
  
  
##### `metricbeat-pythonIntegTest - mage pythonIntegTest`
<ul>
<li>Took 2 min 44 sec . View more details <a href="https://beats-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/Beats/pipelines/beats/pipelines/PR-30399/runs/1/steps/3890/log/?start=0">here</a></li>
<li>Description: <code>mage pythonIntegTest</code></l1>
</ul>
  
  
  
  
##### `metricbeat-windows-windows-2019 - mage build unitTest`
<ul>
<li>Took 5 min 42 sec . View more details <a href="https://beats-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/Beats/pipelines/beats/pipelines/PR-30399/runs/1/steps/3614/log/?start=0">here</a></li>
<li>Description: <code>mage build unitTest</code></l1>
</ul>
  
  
  
  
##### `metricbeat-windows-windows-2019 - mage build unitTest`
<ul>
<li>Took 2 min 24 sec . View more details <a href="https://beats-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/Beats/pipelines/beats/pipelines/PR-30399/runs/1/steps/4006/log/?start=0">here</a></li>
<li>Description: <code>mage build unitTest</code></l1>
</ul>
  
  
  
  
##### `metricbeat-windows-windows-2019 - mage build unitTest`
<ul>
<li>Took 2 min 3 sec . View more details <a href="https://beats-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/Beats/pipelines/beats/pipelines/PR-30399/runs/1/steps/4155/log/?start=0">here</a></li>
<li>Description: <code>mage build unitTest</code></l1>
</ul>
  
  
  
  
##### `Error signal`
<ul>
<li>Took 0 min 0 sec . View more details <a href="https://beats-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/Beats/pipelines/beats/pipelines/PR-30399/runs/1/steps/3762/log/?start=0">here</a></li>
<li>Description: <code>Error 'hudson.AbortException: script returned exit code 1'</code></l1>
</ul>
  
  </p>
  </details>


## :bug: Flaky test report
:grey_exclamation: There are test failures but not known flaky tests.



<details><summary>Expand to view the summary</summary>
<p>

### Genuine test errors [![7](https://img.shields.io/badge/7%20-red)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2FPR-30399/detail/PR-30399/1//tests)
:broken_heart: There are test failures but not known flaky tests, most likely a genuine test failure.

* **Name**: `Build&Test / metricbeat-goIntegTest / TestMemoryService_GetMemoryStats – github.com/elastic/beats/v7/metricbeat/module/docker/memory`
* **Name**: `Build&Test / metricbeat-unitTest / TestMemoryService_GetMemoryStats – github.com/elastic/beats/v7/metricbeat/module/docker/memory`
* **Name**: `Build&Test / metricbeat-windows-windows-2019 / TestMemoryService_GetMemoryStats – github.com/elastic/beats/v7/metricbeat/module/docker/memory`
* **Name**: `Build&Test / metricbeat-pythonIntegTest / test_cpu_fields – metricbeat.module.docker.test_docker.Test`
* **Name**: `Build&Test / metricbeat-pythonIntegTest / test_diskio_fields – metricbeat.module.docker.test_docker.Test`
* **Name**: `Build&Test / metricbeat-pythonIntegTest / test_memory_fields – metricbeat.module.docker.test_docker.Test`
* **Name**: `Build&Test / metricbeat-pythonIntegTest / test_network_fields – metricbeat.module.docker.test_docker.Test`

</p>
</details>



## :robot: GitHub comments

To re-run your PR in the CI, just comment with:



- `/test` : Re-trigger the build.


- `/package` : Generate the packages and run the E2E tests.


- `/beats-tester` : Run the installation tests with beats-tester.


- `run` `elasticsearch-ci/docs` : Re-trigger the docs validation. (use unformatted text in the comment!)




<!--COMMENT_GENERATED_WITH_ID_comment.id-->

ruflin commented:
I removed the backport label on this as features should not be backported only bug fixes https://github.com/elastic/beats/pull/31039 @MichaelKatsoulis Can you have a look at this?

MichaelKatsoulis commented:
> I removed the backport label on this as features should not be backported only bug fixes #31039 @MichaelKatsoulis Can you have a look at this?

My mistake. Thing is that when I added the label in 15th February there was not a `backport-v8.2.0` label, so I just added that.

ruflin commented:
For features you should never add a backport label.
